sound/soc/sof/topology.c

```diff
@@ -38,6 +38,125 @@
 #define TLV_STEP	1
 #define TLV_MUTE	2
 
+/* send pcm params ipc */
+static int ipc_pcm_params(struct snd_sof_widget *swidget, int dir)
+{
+	struct sof_ipc_pcm_params_reply ipc_params_reply;
+	struct snd_sof_dev *sdev = swidget->sdev;
+	struct sof_ipc_pcm_params pcm;
+	int ret = 0;
+
+	memset(&pcm, 0, sizeof(pcm));
+
+	/* set IPC PCM parameters */
+	pcm.hdr.size = sizeof(pcm);
+	pcm.hdr.cmd = SOF_IPC_GLB_STREAM_MSG | SOF_IPC_STREAM_PCM_PARAMS;
+	pcm.comp_id = swidget->comp_id;
+	pcm.params.hdr.size = sizeof(pcm.params);
```

Comment:
I am a little bit concern sending empty params here which means the detect pipeline won't support runtime format/rate change.

Reply:
@keyonjie lets bring it up in the meeting tomorrow

Reply:
fixed now to use runtime PCM params